File: ClinicSystem/src/com/example/clinicsystem/controllers/C_UserType.java
package com.example.clinicsystem.controllers;

import com.example.clinicsystem.helpers.TextFieldValidator;
import com.example.clinicsystem.models.classes.M_Patient;
import com.example.clinicsystem.models.classes.M_Permission;
import com.example.clinicsystem.models.classes.M_User;
import com.example.clinicsystem.models.classes.M_UserType;

import javax.swing.*;
import java.util.List;

public class C_UserType {
    public List request(String requestType, List data) {
        M_UserType userType = new M_UserType();

        switch (requestType) {
            case "C":
                userType.setTypeName(data.get(0).toString());

                for(int i=1; i < data.size(); i++) {
                    M_Permission permission = (M_Permission)data.get(i);
                    userType.getPermissions().add(permission);
                }

                userType.setUserTypeID(userType.addUserType());
                userType.addTypeLink();

                return null;
            case "R":

            case "U":

            case "D":

        }

        return null;
    }

    public boolean isValidUserType(List<JTextField> textFields , List<JLabel> labels) {
        if(TextFieldValidator.validateEmpty(textFields,labels)) {
            return true;
        }

        return false;
    }
}

File: clinic/C_UserType.php
<?php
include_once "M_UserType.php";
include_once "M_Permission.php";

class C_UserType {
    public function request($requestType, $data) {
        $userType = new M_UserType(1, "", null);
		
        switch ($requestType) {
            case "C":
                $userType->setTypeName($data[0]);

                for($i=1; $i < sizeof($data); $i++) {
                    $permission = $data[i];
					array_push($userType->getPermissions(), $permission);
                }

                $userType->setUserTypeID($userType->addUserType());
                $userType->addTypeLink();

                return null;
            case "R":
				if($data == null) {
                    return $userType->getAllUserTypes("");
                } else {
                    return $userType->getAllUserTypes($data[0]);
                }
            case "U":

            case "D":
			
			default:
				return null;
        }
    }

    /*public function isValidUserType($textFields , $labels) {
        if(TextFieldValidator.validateEmpty(textFields,labels)) {
            return true;
        }

        return false;
    }*/
}
?>